demo/index.html

```diff
@@ -45,8 +48,27 @@
     <div>Auto Validation</div>
 
     <paper-input-container auto-validate>
-      <label>letters only</label>
-      <input pattern="[a-zA-Z]*">
+      <label>only type letters</label>
+      <input is="iron-input" pattern="[a-zA-Z]*">
+      <paper-input-error class="add-on">letters only!</paper-input-error>
```

Comment:
In the case of buttons, we ship a compound widget (for example, `paper-icon-button`) composed of several component parts (card, ripple, icon) and optimize for ease of putting the element in the document (you just add a single tag, and configure with attributes).

In this case, we ask the user to configure a compound widget by composing many atomic widgets in their local DOM. This optimizes for modularity and configuration, but is less easy to add to the document.

This seems like a double standard. Both strategies have merit, but as a user of a package of elements, I probably only expect one strategy or another for that entire package. Which of the two strategies is the "correct" way forward for ours?